controller/src/test/java/io/pravega/controller/server/eventProcessor/SerializedRequestHandlerTest.java

```diff
@@ -0,0 +1,184 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.eventProcessor;
+
+import io.pravega.common.concurrent.FutureHelpers;
+import io.pravega.controller.eventProcessor.impl.EventProcessor;
+import io.pravega.shared.controller.event.ControllerEvent;
+import lombok.Data;
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class SerializedRequestHandlerTest {
```

Comment:
Make this class extend `ThreadPooledTestSuite`. This will auto-provide an executor for you and shut it down. No need for all the manual setup and teardown.